src/utils/server.ts

```diff
@@ -0,0 +1,55 @@
+import express from "express";
+import { fileURLToPath } from "url";
+import path from "path";
+import type { Server } from "http";
+import net from "net";
+
+const app = express();
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+let server: Server;
+
+export const runServer = async () => {
+  app.use(express.static(path.join(__dirname, "../../public")));
+  const port = await findPort();
+  server = app.listen(port);
+  return port;
+};
+
+export const closeServer = () => {
```

Comment:
return server.close() in runServer
 whoever calls it will have the responsibilty of closign the server (i,e in an after/afterEach)